### PR TITLE
testsuite refactor

### PR DESCRIPTION
**Pull request by jeenbroekstra**
This PR addresses GitHub issue: eclipse/rdf4j#1236 .

Briefly describe the changes proposed in this PR:

* all storage-related testsuites and compliance tests moved here
* several tests migrated to proper unit tests in their respective modules

To be merged together with PR eclipse/rdf4j#1372 and eclipse/rdf4j-testsuite#58.

After this thas been merged we will need to adapt the jenkins setup for the develop branch.

### CODE REVIEW

**Comment by hmottestad** [comment]
looks like you forgot to sign-off

File: compliance/store/src/test/java/org/eclipse/rdf4j/sail/memory/MemInferencingTest.java [hunk 1]
```diff
@@ -12,12 +12,13 @@
...
 import org.eclipse.rdf4j.sail.Sail;
 import org.eclipse.rdf4j.sail.inferencer.fc.ForwardChainingRDFSInferencer;
+import org.eclipse.rdf4j.sail.inferencer.fc.SchemaCachingRDFSInferencer;
 
 public class MemInferencingTest extends InferencingTest {
 
 	@Override
 	protected Repository createRepository() {
-		Sail sailStack = new ForwardChainingRDFSInferencer(new MemoryStore());
+		Sail sailStack = new SchemaCachingRDFSInferencer(new MemoryStore());
```
Comment (hmottestad) [inline compliance/store/src/test/java/org/eclipse/rdf4j/sail/memory/MemInferencingTest.java]: I'm fairly certian

Comment (hmottestad) [reply compliance/store/src/test/java/org/eclipse/rdf4j/sail/memory/MemInferencingTest.java]: Though we should obviously double check that I didn't forget to duplicate a test ;)

**Comment by hmottestad** [comment]
Safari is crashing when I try to look at the diff, so I will just say: LGTM

**Comment by jeenbroekstra** [comment]
> looks like you forgot to sign-off

Nah, it's a weird bug in the validator which has something to do with how github identifies the author of a merge commit that's the result of a pull request being merged (basically it used some address like  'foobar@github.noreply' instead of my actuall address). The author of the validation plugin is aware and looking into a fix. For now I just check the details and then ignore :)

**Comment by jeenbroekstra** [comment]
@rdf4j-bot run tests

**Comment by jeenbroekstra** [comment]
@rdf4j-bot run tests

File: compliance/store/src/test/java/org/eclipse/rdf4j/sail/memory/MemInferencingTest.java (hunk 1 above)
Comment (jeenbroekstra) [reply compliance/store/src/test/java/org/eclipse/rdf4j/sail/memory/MemInferencingTest.java]: Fair point Havard. I reckon that once this is merged and I have the Jenkins process back on track, we can do a second pass over all the tests, and weed out anything that is unnecessary and/or out of date.

**Pull request merged**